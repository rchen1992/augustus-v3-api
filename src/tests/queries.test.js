const { runQuery } = require('@tests/utils');
const { queryTestCases } = require('./queries');

describe('Queries', () => {
    // Running the test for each case in the cases array.
    queryTestCases.forEach(obj => {
        const { id, query, variables, context, expected } = obj;

        test(`query: ${id}`, async () => {
            const result = await runQuery(query, variables, context);

            /**
             * If we provide an expected result, check against it.
             * Otherwise, just use snapshot testing.
             */
            if (expected) {
                return expect(result).toEqual(expected);
            } else {
                return expect(result).toMatchSnapshot();
            }
        });
    });
});
