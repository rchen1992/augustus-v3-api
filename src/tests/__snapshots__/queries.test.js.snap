// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Queries query: Get all users basic data 1`] = `
Object {
  "data": Object {
    "users": Array [
      Object {
        "avatarUrl": "https://robohash.org/solutaeaut.bmp?size=50x50&set=set1",
        "createdAt": "2018-09-25 03:08:51",
        "email": "mjurczak0@epa.gov",
        "id": "1",
        "updatedAt": "2018-07-12 05:22:33",
        "userName": "cblanko0",
      },
      Object {
        "avatarUrl": "https://robohash.org/nesciuntnobismagni.png?size=50x50&set=set1",
        "createdAt": "2018-08-01 19:26:23",
        "email": "swoodyatt1@storify.com",
        "id": "2",
        "updatedAt": "2018-07-14 22:48:36",
        "userName": "csaye1",
      },
      Object {
        "avatarUrl": "https://robohash.org/cupiditatesedfugiat.jpg?size=50x50&set=set1",
        "createdAt": "2018-09-13 15:51:54",
        "email": "lmckniely2@berkeley.edu",
        "id": "3",
        "updatedAt": "2019-04-06 09:37:00",
        "userName": "cburchell2",
      },
      Object {
        "avatarUrl": "https://robohash.org/molestiaeinciduntlaboriosam.bmp?size=50x50&set=set1",
        "createdAt": "2019-01-01 16:47:52",
        "email": "gbanat3@typepad.com",
        "id": "4",
        "updatedAt": "2018-04-20 21:04:19",
        "userName": "etappor3",
      },
      Object {
        "avatarUrl": "https://robohash.org/voluptatumundemodi.bmp?size=50x50&set=set1",
        "createdAt": "2018-08-16 17:09:23",
        "email": "sdennistoun4@wikimedia.org",
        "id": "5",
        "updatedAt": "2018-08-18 01:01:35",
        "userName": "selcott4",
      },
    ],
  },
}
`;

exports[`User queries query: Get all users basic data 1`] = `
Object {
  "data": Object {
    "users": Array [
      Object {
        "avatarUrl": "https://robohash.org/solutaeaut.bmp?size=50x50&set=set1",
        "createdAt": "2018-09-25 03:08:51",
        "email": "mjurczak0@epa.gov",
        "id": "1",
        "updatedAt": "2018-07-12 05:22:33",
        "userName": "cblanko0",
      },
      Object {
        "avatarUrl": "https://robohash.org/nesciuntnobismagni.png?size=50x50&set=set1",
        "createdAt": "2018-08-01 19:26:23",
        "email": "swoodyatt1@storify.com",
        "id": "2",
        "updatedAt": "2018-07-14 22:48:36",
        "userName": "csaye1",
      },
      Object {
        "avatarUrl": "https://robohash.org/cupiditatesedfugiat.jpg?size=50x50&set=set1",
        "createdAt": "2018-09-13 15:51:54",
        "email": "lmckniely2@berkeley.edu",
        "id": "3",
        "updatedAt": "2019-04-06 09:37:00",
        "userName": "cburchell2",
      },
      Object {
        "avatarUrl": "https://robohash.org/molestiaeinciduntlaboriosam.bmp?size=50x50&set=set1",
        "createdAt": "2019-01-01 16:47:52",
        "email": "gbanat3@typepad.com",
        "id": "4",
        "updatedAt": "2018-04-20 21:04:19",
        "userName": "etappor3",
      },
      Object {
        "avatarUrl": "https://robohash.org/voluptatumundemodi.bmp?size=50x50&set=set1",
        "createdAt": "2018-08-16 17:09:23",
        "email": "sdennistoun4@wikimedia.org",
        "id": "5",
        "updatedAt": "2018-08-18 01:01:35",
        "userName": "selcott4",
      },
    ],
  },
}
`;
